src/chainstate/burn/operations/mod.rs

```diff
@@ -61,86 +106,151 @@ impl error::Error for Error {
         match *self {
             Error::ParseError => None,
             Error::InvalidInput => None,
-            Error::DBError(ref e) => Some(e)
+
+            Error::BlockCommitPredatesGenesis => None,
+            Error::BlockCommitAlreadyExists => None,
+            Error::BlockCommitNoLeaderKey => None,
+            Error::BlockCommitLeaderKeyAlreadyUsed => None,
+            Error::BlockCommitNoParent => None,
+            Error::BlockCommitBadInput => None,
+
+            Error::LeaderKeyAlreadyRegistered => None,
+            Error::LeaderKeyBadConsensusHash => None,
+
+            Error::UserBurnSupportBadConsensusHash => None,
+            Error::UserBurnSupportNoLeaderKey => None,
         }
     }
 
     fn description(&self) -> &str {
         match *self {
             Error::ParseError => "Failed to parse transaction into Blockstack operation",
             Error::InvalidInput => "Invalid input",
-            Error::DBError(ref e) => e.description()
+
+            Error::BlockCommitPredatesGenesis => "Block commit predates genesis block",
+            Error::BlockCommitAlreadyExists => "Block commit commits to an already-seen block",
+            Error::BlockCommitNoLeaderKey => "Block commit has no matching register key",
+            Error::BlockCommitLeaderKeyAlreadyUsed => "Block commit register key already used",
+            Error::BlockCommitNoParent => "Block commit parent does not exist",
+            Error::BlockCommitBadInput => "Block commit tx input does not match register key tx output",
+
+            Error::LeaderKeyAlreadyRegistered => "Leader key has already been registered",
+            Error::LeaderKeyBadConsensusHash => "Leader key has an invalid consensus hash",
+
+            Error::UserBurnSupportBadConsensusHash => "User burn support has an invalid consensus hash",
+            Error::UserBurnSupportNoLeaderKey => "User burn support does not match a registered leader key",
         }
     }
 }
 
-#[derive(Debug, Clone, PartialEq)]
-pub enum CheckResult {
-    BlockCommitOk,
-    BlockCommitPredatesGenesis,
-    BlockCommitBadEpoch,
-    BlockCommitNoLeaderKey,
-    BlockCommitLeaderKeyAlreadyUsed,
-    BlockCommitNoParent,
-    BlockCommitBadInput,
+#[derive(Debug, PartialEq, Clone, Eq)]
+pub struct LeaderBlockCommitOp {
+    pub block_header_hash: BlockHeaderHash, // hash of Stacks block header (double-sha256)
+    pub new_seed: VRFSeed,                  // new seed for this block
+    pub parent_block_ptr: u32,              // pointer to the block that contains the parent block hash 
+    pub parent_vtxindex: u16,               // offset in the parent block where the parent block hash can be found
+    pub key_block_ptr: u32,                 // pointer to the block that contains the leader key registration 
+    pub key_vtxindex: u16,                  // offset in the block where the leader key can be found
+    pub memo: Vec<u8>,                      // extra unused byte
```

Comment:
I'm confused by the typing here: should we have 
```
    pub parent_block_ptr: u64,
    pub parent_vtxindex: u32,
    pub key_block_ptr: u64,
    pub key_vtxindex: u32,
```
instead?

Reply:
In this case, the sizes of these fields correspond to the sizes in the OP_RETURN.